src/hooks/use-menu/index.tsx

```diff
@@ -33,6 +46,48 @@ export const useMenu = (): useMenu => {
     return {
       ...props,
       'data-open': isOpen,
```

Comment:
**note:** No need to do this now, but I want to flag that I think this hook has enough complexity now to warrant splitting out each prop getter to its own module (e.g. a custom hook or factory function) so that useMenu's implementation becomes one of "delegation". To give you and idea of what this _could_ look like (without much thought):

```javascript
function useMenu() {
  const [isOpen, setIsOpen] = useState(false)
  const getTriggerProps = useGetTriggerProps(...)
  const getTriggerProps = useGetPopoverProps(...)
  // and so on...
}
```